parmed/formats/psf.py

```diff
@@ -267,8 +267,42 @@ def improp_gen(struct):
             if len(struct.atoms) % 8 != 0: dest.write('\n')
             dest.write('\n')
             # NUMLP/NUMLPH section
-            dest.write((intfmt*2) % (0, 0) + ' !NUMLP NUMLPH\n')
-            dest.write('\n')
+            # if struct.groups.numlp != None:
+            if hasattr(struct.groups, 'numlp') and struct.groups.numlp != None:
+                dest.write(
+                    (intfmt * 2) % (struct.groups.numlp[0], struct.groups.numlp[1])
+                    + " !NUMLP NUMLPH\n"
+                )
+                last_line = ""
+                i = 1
+                for pos, atom in enumerate(struct.atoms):
+                    if hasattr(atom, "frame"):
+                        dest.write(
+                            (intfmt * 2) % (2, i)
+                            + ("%4s%10.5f%14.5f%14.5f")
+                            % (
+                                "F",
+                                atom.frame.local_position[0],
+                                atom.frame.local_position[1],
+                                atom.frame.local_position[2],
```

Comment:
The local_position for this frame isn't quite right.  It's transformed relative to what is in the PSF file so this doesn't work.

I had to change things a bit to get this functionality to work.  I also added support for writing Drude anisotropy records correctly.  It's reordered relative to what CHARMM puts in the PSF, but I've checked programmatically and it looks to all be correct.  I've added test cases for it as well.